File: greedy/java/src/TasksAndDeadlines.java
import java.util.AbstractMap.SimpleEntry;
import java.util.Arrays;
import java.util.Collections;
import java.util.List;

public class TasksAndDeadlines {
	public static void main(String[] args) {
		System.out.println(maximumReward(3, Arrays.asList(
	            new SimpleEntry<>(6, 10),
	            new SimpleEntry<>(8, 15),
	            new SimpleEntry<>(5, 12)
	    )));
		
		System.out.println(maximumReward(10, Arrays.asList(
	            new SimpleEntry<>(3, 47),
	            new SimpleEntry<>(5, 11),
	            new SimpleEntry<>(1, 70),
	            new SimpleEntry<>(2, 100),
	            new SimpleEntry<>(2, 41),
	            new SimpleEntry<>(2, 66),
	            new SimpleEntry<>(5, 80),
	            new SimpleEntry<>(4, 84),
	            new SimpleEntry<>(5, 81),
	            new SimpleEntry<>(5, 40)
	    )));
		
		System.out.println(maximumReward(10, Arrays.asList(
	            new SimpleEntry<>(80, 55),
	            new SimpleEntry<>(29, 46),
	            new SimpleEntry<>(58, 5),
	            new SimpleEntry<>(92, 80),
	            new SimpleEntry<>(62, 68),
	            new SimpleEntry<>(64, 20),
	            new SimpleEntry<>(78, 56),
	            new SimpleEntry<>(41, 66),
	            new SimpleEntry<>(62, 44),
	            new SimpleEntry<>(32, 80)
	    )));
	}

    public static long maximumReward(int n, List<SimpleEntry<Integer, Integer>> tasks) {
        // Ordena as tarefas pelo primeiro elemento do par
        Collections.sort(tasks, (a, b) -> a.getKey().compareTo(b.getKey()));

        long reward = 0, time = 0;
        for (SimpleEntry<Integer, Integer> task : tasks) {
            time += task.getKey();
            reward += (task.getValue() - time);
        }

        return reward;
    }
}

File: greedy/python/tasks_and_deadlines.py
class Solution:
    def maximum_reward(self, tasks):
        # Ordena as tarefas por duração e prazo
        tasks.sort()

        reward = 0
        time = 0
        for duration, deadline in tasks:
            time += duration
            reward += (deadline - time)

        return reward

s = Solution()
input1 = [[6, 10],
          [8, 15],
          [5, 12]]
print(s.maximum_reward(input1))

input2 = [[3, 47],
          [5, 11],
          [1, 70],
          [2, 100],
          [2, 41],
          [2, 66],
          [5, 80],
          [4, 84],
          [5, 81],
          [5, 40]]
print(s.maximum_reward(input2))

input3 = [[80, 55],
          [29, 46],
          [58, 5],
          [92, 80],
          [62, 68],
          [64, 20],
          [78, 56],
          [41, 66],
          [62, 44],
          [32, 80]]
print(s.maximum_reward(input3))